addons/survey/static/src/js/survey_form.js

```diff
@@ -108,6 +110,10 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
         if (this.$('fieldset[disabled="disabled"]').length !== 0) {
             return;
         }
+        // Disable all navigation keys when zoom modal is open, but the ESC.
+        if ($('.o_survey_img_zoomer_body').length && event.keyCode !== 27) {
```

Comment:
Could be better to check for something like `if (this.surveyImageZoom)` or something rather than a global selector.

Reply:
Done. I've used a small event to tell the form when the modal is closed.